src/taco-dependency-installer/installers/androidPackagesInstaller.ts

```diff
@@ -77,26 +82,25 @@ class AndroidPackagesInstaller extends InstallerBase {
         }
     }
 
-    private getAvailableAndroidPackages(): Q.Promise<any> {
-        var deferred: Q.Deferred<any> = Q.defer();
+    private executeAndroidCommand(name: string, args: string[]): IExecuteCommandResult {
```

Comment:
Also, small nit: it took me a few moments to realize that the `name` parameter was just for telemetry purposes. I think it would be good to make it more obvious. I would put it as second parameter, and make it optional (default to args[0]). That way, consumers can invoke Android commands without worrying about telemetry-specific stuff. I would also change the name slightly to reflect better that it's a telemetry param.

So, this is what I propose (combined with my above comment):
```typescript
private executeAndroidCommand(args: string[], telemetryStep?: string): Q.Promise<string> {
    if(!telemetryStep) {
        telemetryStep = args[0]; // You may have to deal with null / empty args?
    }
    // ...
}
```